pkg/knowledge_base2/kb.go

```diff
@@ -12,6 +12,7 @@ import (
 )
 
 //go:generate mockgen --source=./kb.go -destination=./template_kb_mock_test.go -package=knowledgebase2
+//go:generate mockgen --source=./kb.go -destination=../engine2/operational_eval/template_kb_mock_test.go -package=operational_eval
```

Comment:
This is really awkward - should just generate one mock. That's why there's the `graphtest`, `enginetest`, `kbtesting` etc packages. It's also a little awkward, but it's standard for Go for reusable test utilities (see for eg. [httptest](https://pkg.go.dev/net/http/httptest) in the stdlib).

Reply:
well get into the circular dependencies again. Theres no way to test the actual solution context package for example. but we could try a different method, i just think this gets rid of that worry